src/adaptations/device-layer/include/Weave/DeviceLayer/internal/GenericConfigurationManagerImpl.ipp

```diff
@@ -88,8 +89,29 @@ WEAVE_ERROR GenericConfigurationManagerImpl<ImplClass>::_ConfigureWeaveStack()
     FabricState.GroupKeyStore = Impl()->_GetGroupKeyStore();
 
 #if WEAVE_PROGRESS_LOGGING
+
     Impl()->LogDeviceConfig();
-#endif
+
+#if WEAVE_DEVICE_CONFIG_LOG_PROVISIONING_HASH
+    {
+        uint8_t provHash[Platform::Security::SHA256::kHashLength];
+        char provHashBase64[BASE64_ENCODED_LEN(sizeof(provHash)) + 1];
+        err = Impl()->_ComputeProvisioningHash(provHash, sizeof(provHash));
+        if (err == WEAVE_NO_ERROR)
+        {
+            Base64Encode(provHash, sizeof(provHash), provHashBase64);
+            provHashBase64[sizeof(provHashBase64) - 1] = 0;
```

Comment:
Style: minor issue; however, I find that code that follows "= '\0'" makes it more clear that NULL termination is being performed.